nucypher-core-wasm/src/lib.rs

```diff
@@ -536,6 +536,104 @@ impl EncryptedTreasureMap {
     }
 }
 
+//
+// Threshold Decryption Request
+//
+
+#[wasm_bindgen]
+#[derive(PartialEq, Eq, Debug, derive_more::From, derive_more::AsRef)]
+pub struct ThresholdDecryptionRequest(nucypher_core::ThresholdDecryptionRequest);
+
+#[wasm_bindgen]
+impl ThresholdDecryptionRequest {
+    #[wasm_bindgen(constructor)]
+    pub fn new(
+        id: u16,
+        variant: u8,
+        ciphertext: &[u8],
+        conditions: &OptionConditions,
+        context: &OptionContext,
+    ) -> Result<ThresholdDecryptionRequest, Error> {
+        let typed_conditions = try_from_js_option::<Conditions>(conditions)?;
+        let typed_context = try_from_js_option::<Context>(context)?;
+
+        let ferveo_variant = match variant {
+            0 => FerveoVariant::SIMPLE,
+            1 => FerveoVariant::PRECOMPUTED,
+            _ => return Err(Error::new("Invalid variant")),
+        };
+
+        Ok(Self(nucypher_core::ThresholdDecryptionRequest::new(
+            id,
+            ciphertext,
+            typed_conditions.as_ref().map(|conditions| &conditions.0),
+            typed_context.as_ref().map(|context| &context.0),
+            &ferveo_variant,
+        )))
+    }
+
+    #[wasm_bindgen(getter)]
+    pub fn id(&self) -> u16 {
+        self.0.ritual_id
+    }
+
+    #[wasm_bindgen(getter)]
+    pub fn variant(&self) -> u8 {
+        match self.0.variant {
+            FerveoVariant::SIMPLE => 0,
+            FerveoVariant::PRECOMPUTED => 1,
+        }
+    }
+
+    #[wasm_bindgen(getter)]
+    pub fn ciphertext(&self) -> Box<[u8]> {
+        self.0.ciphertext.clone()
+    }
+
+    #[wasm_bindgen(js_name = fromBytes)]
+    pub fn from_bytes(data: &[u8]) -> Result<ThresholdDecryptionRequest, Error> {
+        from_bytes::<_, nucypher_core::ThresholdDecryptionRequest>(data)
+    }
+
+    #[wasm_bindgen(js_name = toBytes)]
+    pub fn to_bytes(&self) -> Box<[u8]> {
+        to_bytes(self)
+    }
+}
+
+//
+// Threshold Decryption Response
+//
+
+#[wasm_bindgen]
+#[derive(PartialEq, Eq, Debug, derive_more::From, derive_more::AsRef)]
+pub struct ThresholdDecryptionResponse(nucypher_core::ThresholdDecryptionResponse);
+
+#[wasm_bindgen]
+impl ThresholdDecryptionResponse {
+    #[wasm_bindgen(constructor)]
+    pub fn new(decryption_share: &[u8]) -> Result<ThresholdDecryptionResponse, Error> {
+        Ok(Self(nucypher_core::ThresholdDecryptionResponse::new(
+            decryption_share.into(),
+        )))
+    }
+
+    #[wasm_bindgen(getter)]
+    pub fn decryption_share(&self) -> Box<[u8]> {
```

Comment:
Needs `js_name = decryptionShare`, to match JS style.

Reply:
✔️